src/application-development/alloc.md

```diff
@@ -0,0 +1,59 @@
+# Alloc
+
+In a `no_std` environment, the `alloc` crate is available as an option for heap allocation. It can be useful when working with crates that require alloc or when using dynamic collections like `Vec`.
+
+We provide our own `no_std` heap allocator [esp-alloc]. To use it you need to:
+
+1. Add a dependency to your `Cargo.toml`
+```toml
+esp-alloc        = "0.7.0"
+```
+
+2. Modify the `.cargo/config.toml`
+```toml
+[unstable]
+build-std = ["alloc", "core"] # added alloc here
+```
+
+3. Use `alloc` crate in your application
+```rust
+// main.rs
+
+extern crate alloc;
+```
+
+4. Initialize a global heap allocator providing a heap of the given size in bytes with the provided macro:
+```rust
+// main.rs
+
+esp_alloc::heap_allocator!(size: 72 * 1024);
+```
+
+or when more customization is required using the function:
+```rust
+// main.rs
+
+use esp_alloc as _;
+fn init_heap() {
+    const HEAP_SIZE: usize = 32 * 1024;
+    static mut HEAP: MaybeUninit<[u8; HEAP_SIZE]> = MaybeUninit::uninit();
+    unsafe {
+        esp_alloc::HEAP.add_region(esp_alloc::HeapRegion::new(
+            HEAP.as_mut_ptr() as *mut u8,
+            HEAP_SIZE,
+            esp_alloc::MemoryCapability::Internal.into(),
+        ));
+    }
+}
+```
+
+## Configurable Memory Placement and Reclaimed RAM
+
+On chips, like the esp32 with non-contiguous memory, `cfg` options can be used to control where memory is placed. This also allows accessing RAM that is otherwise unavailable, such as memory occupied by the 2nd stage bootloader. For supported chips, regions like `.dram2_uninit` can be used as additional heap memory, optimizing available resources.
```

Comment:
IMHO no need to call out `ESP32` specifically

Reply:
And if we do, we should use proper capitalization, i.e. ESP32